### PR TITLE
linker: Add ROM_SECTIONS location

### PR DESCRIPTION
**Pull request by pdgendt**
Allow putting sections after `common-rom`.

### CODE REVIEW

File: include/zephyr/arch/arm/cortex_m/scripts/linker.ld [hunk 1]
```diff
@@ -187,6 +187,7 @@ SECTIONS
 	__rodata_region_start = .;
 
 #include <zephyr/linker/common-rom.ld>
+#include <snippets-rom-sections.ld>
```
Comment (ithinuel) [inline include/zephyr/arch/arm/cortex_m/scripts/linker.ld:190]: Like for the other occcurences of snippet, this should be documented with something of the form
```
/* Located in generated directory. This file is populated by calling
 * zephyr_linker_sources(ROM_SECTIONS ...). <description of typical use>.
 */
#include <snippets-rom-sections.ld>
```

**Review by ithinuel — CHANGES_REQUESTED** [review]

**Review by dcpleung — DISMISSED** [review]

**Review by gmarull — DISMISSED** [review]

**Comment by pdgendt** [comment]
I actually no longer have a dependency on this, but I guess it doesn't harm either?
In-tree there are many uses of `zephyr_linker_sources(SECTIONS ...)`, which I thought had the behavior I was expecting like the `ROM_SECTIONS` in this PR.

**Comment by pdgendt** [comment]
Ping @ithinuel is comment addressed?

**Review by ithinuel — DISMISSED** [review]

**Comment by nashif** [comment]
> Allow putting sections after common-rom.

More details in the commit message following https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-message-body are appreciated, This is very short and missing the why and any further details.

**Comment by pdgendt** [comment]
@nashif I've updated the commit message

@tejlmand does the `SECTIONS` put after bss/noinit make sense to begin with? Shouldn't it be always in ROM? Ref #72307

**Review by nashif — APPROVED** [review]

**Comment by pdgendt** [comment]
Assigned to @tejlmand to get feedback

**Review by tejlmand — APPROVED** [review]

**Pull request merged**